### PR TITLE
 v1: fix the response data from the recommendation service

### PR DESCRIPTION
**Pull request by mgold1234**
Unmarshal the response into an appropriate struct.

### CODE REVIEW

File: internal/v1/handler_recommend_package.go [hunk 1]
```diff
@@ -39,16 +39,19 @@ func (h *Handlers) handleRecommendationsResponse(ctx echo.Context, uploadPackage
 
 	ctx.Logger().Debugf("Getting Response list of items %v", resp)
 	defer closeBody(ctx, resp.Body)
-	var packages []string
-	err = json.NewDecoder(resp.Body).Decode(&packages)
+	type ResponsePackages struct {
```
Comment (croissanne) [inline internal/v1/handler_recommend_package.go:42]: No need to define a new struct, just use ```RecommendationsResponse```

File: internal/v1/handler_recommend_package.go [hunk 2]
```diff
@@ -39,16 +39,19 @@ func (h *Handlers) handleRecommendationsResponse(ctx echo.Context, uploadPackage
...
 	}
 
-	if len(packages) == 0 {
+	if len(responsePackages.Packages) == 0 {
 		ctx.Logger().Errorf("User should define packages")
 		return RecommendationsResponse{}, nil
 	}
 
-	return RecommendationsResponse{Packages: packages}, nil
+	return RecommendationsResponse{Packages: responsePackages.Packages}, nil
```
Comment (croissanne) [inline internal/v1/handler_recommend_package.go:56]: Just return the RecommendationsResponse into which you unmarshal the response.

**Review by croissanne — CHANGES_REQUESTED** [review]

**Review by croissanne — CHANGES_REQUESTED** [review]
Thanks! Could you also:
- squash the commits
- fix the typo in the commit title
- max length of the commit title is 65
- set the module to something like `internal/v1`
- construct a RecommendationsResponse only once instead of twice.

**Review by croissanne — CHANGES_REQUESTED** [review]
Can you reword the commit message?
Suggestion:
```
 v1: fix the response data from the recommendation service

Unmarshal the response into an appropriate struct.
```

**Review by croissanne — APPROVED** [review]

**Review by lzap — APPROVED** [review]
LGTM

**Pull request merged**